app/src/main/java/org/zephyrsoft/trackworktime/timer/TimeCalculatorV2.java

```diff
@@ -361,7 +374,7 @@ public void calculateNextDay() {
 
 		currentDayHasEvents = (events != null && !events.isEmpty());
 
-		workedTime += calculateWorkTime(events);
+		workedTime += roundUpToMultiple(calculateWorkTime(events), smallestTimeUnit);
```

Comment:
This should be wrapped in an `if` block to only execute it when required:
```
if (roundingEnabled) {
   workedTime += roundUpToMultiple(calculateWorkTime(events), smallestTimeUnit);
}
```

Reply:
While glancing over this again, my previous comment doesn't make sense. The `calculateWorkTime(events)` would also be only executed if my code snippet was used, so it better should be like:

```
if (roundingEnabled) {
	workedTime += roundUpToMultiple(calculateWorkTime(events), smallestTimeUnit);
} else {
	workedTime += calculateWorkTime(events);
}
```